examples/asrc_demo/src/rate_server.h

```diff
@@ -4,33 +4,48 @@
 #define RATE_SERVER_H
 #include "xmath/xmath.h"
 
+#define CHECK_SAMPLES_TO_HOST_BUF_WRITE_TIME (0)
+
 void rate_server(void *args);
-void calc_avg_i2s_send_buffer_level(int current_level, bool reset);
 
 // Getters and setters for various global variables
-uint32_t get_i2s_to_usb_rate_ratio();
-void set_i2s_to_usb_rate_ratio(uint32_t ratio);
+uint64_t get_i2s_to_usb_rate_ratio();
+void set_i2s_to_usb_rate_ratio(uint64_t ratio);
 bool get_spkr_itf_close_open_event();
 void set_spkr_itf_close_open_event(bool event);
 
 // Rate calculation math functions
 uint32_t sum_array(uint32_t * array_to_sum, uint32_t array_length);
 float_s32_t float_div(float_s32_t dividend, float_s32_t divisor);
 uint32_t float_div_fixed_output_q_format(float_s32_t dividend, float_s32_t divisor, int32_t output_q_format);
+uint64_t float_div_u64_fixed_output_q_format(float_s32_t dividend, float_s32_t divisor, int32_t output_q_format);
+
+// Wrapper functions for calculating i2s send buffer average level
+void init_calc_i2s_buffer_level_state(void);
+void calc_avg_i2s_send_buffer_level(int32_t current_buffer_level, bool reset);
 
 typedef struct
 {
     /* data */
+    int64_t buffer_based_correction;
     float_s32_t usb_data_rate;
     int32_t samples_to_host_buf_fill_level;
+#if CHECK_SAMPLES_TO_HOST_BUF_WRITE_TIME
```

Comment:
why is this check here? it is hardcoded to zero above, can you remove the whole section?